tests/integration/api/conftest.py

```diff
@@ -270,6 +272,56 @@ def new_activation(
     )
 
 
+@pytest.fixture
+def default_activation_instances(
+    default_activation: models.Activation, default_project: models.Project
+) -> models.RulebookProcess:
+    """Return a list of Activation Instances"""
+    return models.RulebookProcess.objects.bulk_create(
```

Comment:
I think one of our problems is that we rely on models instead of objects created by the api. Through the model, we skip the logic in the serializer and with bulk we skip logic in the save method. Have you considered to create the objects through api requests?

Reply:
I agree. But I actually don't have much info on what the standard conventions are regarding using fixtures for object creations in tests. I know we already do what you are suggesting in e2e tests because that's the only way in e2e tests. Anyway, I'm open to going that way if others also deem it more reasonable.

Reply:
As with many things, it depends. 

1. If a test is exercising a class behavior it's reasonable to use the class directly avoiding potential issues from other source.
2. If a test is exercising a higher-level behavior that requires an instance of a particular class creating the instance at the level of that behavior (assuming that is where it is normally created) is reasonable by the same rationale as in #1.  
3. Any other creation of an instance would be done as in production, as it would be assumed to work, providing the benefit of early detection of issues.

Of course, the above isn't meant to be a straitjacket.  If there are issues with code that is not under test getting in the way of progress relaxing the above may be the correct thing to do.  Though it should be noted so that it can later be "corrected."

Reply:
Making changes to adhere to what I described would not be a short-term effort. 🤕

Reply:
Agree with all. @Dostonbek1 It should not be a effort to address everything. We can split it and iterate. But I think it can be a good opportunity to define the foundations. I think we can make https://issues.redhat.com/browse/AAP-15713 a epic so we can create multiple tickets there. 

Regarding models I would say that by default, we always want the object created through the request. There may be exceptions, but I think there will be less expections in that way rather than models.

Reply:
@jshimkus-rh @Alex-Izquierdo Thank you both for feedback/thoughts. I have reached a point where I refactored most of the test modules to use fixtures and all tests are now passing. In some of the modules, I made changes just enough for tests to pass and left comments as needed for future refactoring; I wanted to avoid making this PR bigger than it already is and we can work on those test modules in future iterations. Please, give this PR another look. :)